experimental/dds/tree2/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -712,43 +715,64 @@ export class ModularChangeFamily
 	}
 
 	public rebaseAnchors(anchors: AnchorSet, over: ModularChangeset): void {
-		anchors.applyDelta(this.intoDelta(over));
+		anchors.applyDelta(this.intoDelta(over, () => EmptyKey));
 	}
 
-	public intoDelta(change: ModularChangeset): Delta.Root {
+	public intoDelta(
+		change: ModularChangeset,
+		repairDataHandler: RepairDataHandler,
+		repairDataMarks = new Map<FieldKey, Delta.MarkList>(),
+	): Delta.Root {
 		// Return an empty delta for changes with constraint violations
 		if ((change.constraintViolationCount ?? 0) > 0) {
 			return new Map();
 		}
 
-		return this.intoDeltaImpl(change.fieldChanges);
+		const repairDataBuilder = {
+			handler: repairDataHandler,
+			marks: repairDataMarks,
+		};
+		const idAllocator = idAllocatorFromMaxId(change.maxId);
+		const delta = this.intoDeltaImpl(change.fieldChanges, repairDataBuilder, idAllocator);
+
+		repairDataBuilder.marks.forEach((marks, field) => delta.set(field, marks));
```

Comment:
Instead of first collecting the marks in a separate map and then adding them all to the root delta, it make sense to directly add them to the root delta (see also the comment I added on RepairDataBuilder.marks).